tests/test_http.py

```diff
@@ -613,6 +636,322 @@ def test_write(self):
             self.assertIsNone(path_expect.write(data=data))
 
 
+class HttpReadWriteWebdavServerTestCase(GenericReadWriteTestCase, unittest.TestCase):
+    """Test with a real webDAV server, as opposed to mocking responses."""
+
+    scheme = "http"
+
+    @classmethod
+    def setUpClass(cls):
+        cls.webdav_tmpdir = tempfile.mkdtemp(prefix="webdav-server-test-")
+        cls.local_files_to_remove = []
+
+        # If a WsgiDAVApp is available, launch a local server to test against
+        if WsgiDAVApp is not None:
+            # Get an available port for the server to listen to
+            cls.port_number = cls._get_port_number()
+
+            cls.stop_webdav_server = False
+            cls.server_thread = Thread(
+                target=cls._serve_webdav,
+                args=(cls, cls.webdav_tmpdir, cls.port_number, lambda: cls.stop_webdav_server),
+                daemon=True,
+            )
+            cls.server_thread.start()
+
+            # Wait for it to start
+            time.sleep(1)
+
+            # Initialize the server endpoint
+            cls.netloc = f"127.0.0.1:{cls.port_number}"
+        elif (test_endpoint := os.getenv("LSST_RESOURCES_HTTP_TEST_SERVER_URL")) is not None:
+            # A test endpoint is configured via the environment. Run this test
+            # case against it.
+            #
+            # This is convenient for testing against real servers in the
+            # developer environment by initializing the environment variable
+            # with the URL of the server, e.g.
+            #    https://dav.example.org:1234/path/to/top/dir
+            u = urllib.parse.urlparse(test_endpoint)
```

Comment:
The intention was to get a clean path, to avoid potential issues with super-sensitive webDAV servers:

```python
>>> p = ResourcePath("https://host.example.org:12345////not/very////clean/path///")
>>> p.path
'////not/very////clean/path///'
```

By a "clean path" I mean a path in which each component is separated by a single `/`. A path with several separators is syntactically correct, though.

I reverted to use `ResourcePath` for parsing.